src/run/ongoing.ts

```diff
@@ -0,0 +1,37 @@
+import { ApiClient } from "../client/apiClient";
+import { StartedRun } from "./start";
+import { Assertion, RunInformationResponse } from "../client/responses/runInformationResponse";
+import { setTimeout } from "timers/promises";
+import * as core from "@actions/core";
+import { isRunning, statusName } from "./status";
+
+export interface FinishedRun {
+  runId: String;
+  statusCode: number;
+  statusName: string;
+  assertions: Assertion[];
+}
+
+export const waitForRunEnd = async (client: ApiClient, startedRun: StartedRun): Promise<FinishedRun> => {
+  await setTimeout(1000); // add 1s delay because run duration might not be populated yet
+  let runInfo: RunInformationResponse;
+  let oldStatus: number = -1;
+  while (true) {
+    runInfo = await client.getRunInformation(startedRun.runId);
+    runInfo.status !== oldStatus ? core.info(statusMsg(runInfo.status)) : core.debug(statusMsg(runInfo.status));
+    oldStatus = runInfo.status;
+    if (isRunning(runInfo.status)) {
+      await setTimeout(5000);
```

Comment:
If you change to have always this timeout at the first in the loop, there will be no need for the specific delay at first in the method.

Reply:
Mmmh... Yes, not really an issue to wait 4 more seconds for the first summary anyway.